calculator.js

```diff
@@ -0,0 +1,47 @@
+const operators = ['+', '-', '*', '/'];
+
+exports.create_calculator = function (operator) {
+  const numbers = [];
+
+  return {
+    add_number: function (number) {
+      numbers.push(number);
+    },
+    calculate: function () {
+      // Return zero if there are no numbers to calculate
+      if (numbers.length === 0) {
+        throw 'Cannot calculate a result because no numbers were given';
+      }
+
+      let answer = numbers[0];
+
+      let i = 0;
+      numbers.forEach(function (number) {
+        if (i === 0) {
+          i++;
+          return;
+        }
+
+        switch (operator) {
```

Comment:
In this loop and switch statement you're accumulating the value of the calculation in the `number` variable and adding/subtracting/multiplying/dividing by each number in turn. There is an array function in javascript called [reduce](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/reduce) which you could use to achieve the same result and not have to worry about setting `answer = numbers[0]` outside of the for loop and then skipping the loop iteration when `i` === 0 . 

You could adapt your switch statement in your code to select a callback function (which uses the appropriate operator) that would be passed as an argument into `numbers.reduce()`

Reply:
Sick